parsing/parser.mly

```diff
@@ -1164,8 +1165,8 @@ module_expr:
       { unclosed "struct" $loc($1) "end" $loc($4) }
   | FUNCTOR attrs = attributes args = functor_args MINUSGREATER me = module_expr
       { wrap_mod_attrs ~loc:$sloc attrs (
-          List.fold_left (fun acc (x, mty) ->
-            mkmod ~loc:$sloc (Pmod_functor (x, mty, acc))
+          List.fold_left (fun acc (x, mty, startpos) ->
+            mkmod ~loc:(startpos, $endpos) (Pmod_functor (x, mty, acc))
```

Comment:
I'm not sure that the code before that would break any of the Refis laws as formulated. Could you be clearer on which principle should be respected? Maybe the laws should be refine.

Previously we would generate nested functors, all with the same location. This should be fine for the first law, each functor is a direct descendant of the previous one and their locations are included (indeed they are identical). It should be fine for the second law, as no two functors have the same direct parent.

What is the property that you are restoring here?

(Another approach could be to make only the first functor non-ghost, and all other functors ghost. Do we have principles to tell us which choice is better?)

Reply:
> I'm not sure that the code before that would break any of the laws as formulated

It broke the second law, if you look at the example I added to the testsuite:
```ocaml
module F = functor (A : S) (B : S) -> struct end;;
```
then `(A : S)` (the argument) and `functor (B : S) -> struct end` (the child functor) are siblings, but their locations overlap (actually, the location of the second covers the location of the first)

Reply:
Ah! Thanks.